tuna/worker_interface.py

```diff
@@ -177,78 +185,102 @@ def queue_end_reset(self):
     with self.bar_lock:
       self.end_jobs.value = 0
 
-  def load_job_queue(self, session, ids):
-    """load job_queue with info for job ids"""
-    # pylint: disable=comparison-with-callable
-    job_cfgs = session.query(self.dbt.job_table, self.dbt.config_table)\
-      .filter(self.dbt.job_table.valid == 1)\
-      .filter(self.dbt.job_table.session == self.dbt.session.id)\
-      .filter(self.dbt.config_table.id == self.dbt.job_table.config)\
-      .filter(self.dbt.job_table.id.in_(ids)).all()
-    # pylint: enable=comparison-with-callable
+  def check_jobs_found(self, job_rows, find_state, imply_end):
+    """check for end of jobs"""
+    if not job_rows:
+      # we are done
+      self.logger.warning('No %s jobs found, session %s', find_state,
+                          self.session_id)
+      if imply_end:
+        self.logger.warning("set end")
+        self.end_jobs.value = 1
+      return False
+    return True
 
-    if len(ids) != len(job_cfgs):
-      raise Exception(
-          f'Failed to load job queue. #ids: {len(ids)} - #job_cgfs: {len(job_cfgs)}'
-      )
-    for job, config in job_cfgs:
-      self.job_queue.put((job, config))
+  def get_job_table(self, job_tuple):
+    """find job table in a job tuple"""
+    try:
+      for i, tble in enumerate(job_tuple):
+        if isinstance(tble, self.dbt.job_table):
+          job_i = i
+          break
+      job_table = job_tuple[job_i]
+    except TypeError:
+      job_table = job_tuple
+
+    return job_table
+
+  def get_job_tables(self, job_rows):
+    """find job tables in query results"""
+    if isinstance(job_rows[0], self.dbt.job_table):
+      job_tables = job_rows
+    else:
+      job_i = 0
+      for i, tble in enumerate(job_rows[0]):
+        if isinstance(tble, self.dbt.job_table):
+          job_i = i
+          break
+      job_tables = [row[job_i] for row in job_rows]
+
+    return job_tables
+
+  def refresh_query_objects(self, session, rows):
+    """refresh objects in query rows"""
+    for obj_tuple in rows:
+      try:
+        for entry in obj_tuple:
+          session.refresh(entry)
+      except TypeError:
+        session.refresh(obj_tuple)
```

Comment:
Why not just refresh the entire tuple everytime ?

Reply:
I was under the impression that the db objects, had to be refreshed individually. If not then that would be a good solution.

Reply:
Refreshing the tuple doesn't appear to be possible.